File: data/src/main/java/com/zxjdev/smile/data/entity/MomentEntity.java
package com.zxjdev.smile.data.entity;

import com.avos.avoscloud.AVClassName;
import com.avos.avoscloud.AVObject;

@AVClassName("Moment")
public class MomentEntity extends AVObject {

    public static final transient Creator CREATOR;

    static {
        CREATOR = AVObjectCreator.instance;
    }

    public MomentEntity() {
        super();
    }

    public void setContent(String content) {
        put("content", content);
    }

    public String getContent() {
        return getString("content");
    }

    public void setOwner(UserEntity user) {
        put("user", user);
    }

    public UserEntity getUser() {
        return getAVUser("user", UserEntity.class);
    }
}

File: app-data/src/main/java/com/zxjdev/smile/data/moment/entity/MomentEntity.kt
package com.zxjdev.smile.data.moment.entity

import com.avos.avoscloud.AVClassName
import com.avos.avoscloud.AVObject
import com.zxjdev.smile.data.user.entity.UserEntity

@AVClassName("Moment")
class MomentEntity : AVObject() {

    var content: String?
        get() = getString("content")
        set(content) = put("content", content)

    val user: UserEntity
        get() = getAVUser("user", UserEntity::class.java)

    fun setOwner(user: UserEntity) {
        put("user", user)
    }
}
